File: src/main/java/com/lessmarkup/userinterface/model/global/UserGroupModel.java
package com.lessmarkup.userinterface.model.global;

import com.lessmarkup.TextIds;
import com.lessmarkup.dataobjects.UserGroup;
import com.lessmarkup.framework.data.RecordModelWithEditableCollection;
import com.lessmarkup.interfaces.recordmodel.InputField;
import com.lessmarkup.interfaces.recordmodel.InputFieldType;
import com.lessmarkup.interfaces.recordmodel.RecordColumn;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

@Component
@Scope("prototype")
public class UserGroupModel extends RecordModelWithEditableCollection<UserGroupModel, UserGroup> {

    private String name;
    private String description;

    public UserGroupModel() {
        super(TextIds.GROUP, UserGroup.class, UserGroupModel.class);
    }

    @RecordColumn(textId = TextIds.NAME)
    @InputField(textId = TextIds.NAME, type = InputFieldType.TEXT, required = true)
    public void setName(String name) { this.name = name; }
    public String getName() { return name; }

    @RecordColumn(textId = TextIds.DESCRIPTION)
    @InputField(textId = TextIds.DESCRIPTION, type = InputFieldType.TEXT)
    public void setDescription(String description) { this.description = description; }
    public String getDescription() { return description; }
}

File: src/main/scala/com/lessmarkup/userinterface/model/global/UserGroupModel.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.userinterface.model.global

import com.lessmarkup.TextIds
import com.lessmarkup.dataobjects.UserGroup
import com.lessmarkup.framework.data.RecordModelWithEditableCollection
import com.lessmarkup.interfaces.annotations.{InputField, InputFieldType, RecordColumn}

class UserGroupModel extends RecordModelWithEditableCollection[UserGroupModel, UserGroup](TextIds.GROUP, classOf[UserGroup], classOf[UserGroupModel]) {

  @RecordColumn(textId = TextIds.NAME)
  @InputField(textId = TextIds.NAME, fieldType = InputFieldType.TEXT, required = true)
  var name: String = null
  @RecordColumn(textId = TextIds.DESCRIPTION)
  @InputField(textId = TextIds.DESCRIPTION, fieldType = InputFieldType.TEXT)
  var description: String = null
}
